### DISCUSSION TITLE
Throwing 404 from getStaticProps fallback

### DISCUSSION THREAD

**Discussion by bradders** [vercel/next.js#11464] [Help]
Is it possible to throw a 404 (with the correct status code) from inside the fallback of getStaticProps?

The page works as expected: it renders a list of 10 forecasts at build time, and the fallback loads any that aren't pre rendered, and displays the error page when none are found. But when the error page is displayed, it doesn't give a 404 response, it's all 200 OK.

Maybe this is a misuse of getStaticProps, and is an example of when to switch to SSR instead?

```
import fetch from "node-fetch";
import { useRouter } from "next/router";

import ErrorPage from "next/error";
import Layout from "components/Layout.js";

function Forecast({ forecast }) {
  const router = useRouter();

  if (forecast && forecast.error != undefined) {
    return <ErrorPage statusCode={404} />;
  }

  const buildOutput = props => {
    if (router.isFallback) {
      return <div>Loading...</div>;
    }

    return (
      <>
        <h1>{forecast.title}</h1>
        <p>...</p>
      </>
    );
  };

  return (
    <>
      <Layout>{buildOutput()}</Layout>
    </>
  );
}

export async function getStaticPaths() {
  const res = await fetch("https://...forecasts");
  const forecasts = await res.json();

  const paths = forecasts.map(forecast => `/forecasts/${forecast.hash}`);

  return { paths, fallback: true };
}

export async function getStaticProps({ params }) {
  const res = await fetch(`https://...forecasts/${params.hash}`);
  const forecast = await res.json();

  return { props: { forecast } };
}

export default Forecast;
```

**Comment by timneutkens** [comment]
You can reference this: https://github.com/zeit/next.js/discussions/10960#discussioncomment-1201

TLDR: the status code does not matter, you have to correctly provide the robots meta tag though.

**Reply by bradders** [reply]
Awesome, thanks for the link. Good to know statics are always 200 and not to worry about the code.

**Reply by MunnyReol** [reply]
just return this `notFound` as `true`.

```
return {
   notFound: true,
}
```

**Reply by doostinharrell** [reply]
To resolve the issue on my end I had to set the fallback value to `false` in getStaticPaths instead of `blocking`.

**Reply by MunnyReol** [reply]
Update to next js 14
